repair/repair.cc

```diff
@@ -605,7 +605,16 @@ void repair::shard_repair_task_impl::check_failed_ranges() {
 }
 
 void repair::shard_repair_task_impl::check_in_abort_or_shutdown() {
-    _as.check();
+    try {
+        _as.check();
+    } catch (...) {
```

Comment:
No need to catch the exception here. You can do instead:
```c++
if (_as.abort_requested() && !_aborted) {
   [...]
}
_as.check();
```

Reply:
Hmm, with the above version, on normal path (fast path), we will do the extra if check all the time. With current version, we will only do _as.check().  Isn't current version better? The extra throw is not very important because it is rare to abort.